PolyEngine/Core/Src/SafePtr.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "Defines.hpp"
+#include "SafePtrRoot.hpp"
+
+namespace Poly
+{
+	template<typename T>
+	class SafePtr : public BaseObject<>
+	{
+	public:
+		SafePtr(SafePtrRoot* pointer)
+		{
+			PointerIdx = SafePtrRoot::RegisterPointer(pointer);
+		}
+
+		SafePtrRoot *operator->()
+		{
+			return SafePtrRoot::GetPointer(PointerIdx);
+		}
+
+		SafePtrRoot *operator*()
+		{
+			return SafePtrRoot::GetPointer(PointerIdx);
+		}
+	private:
```

Comment:
This API is not complete. We need eq and neq comparison operators (with `SafePtr<T>` and with `T*`).  Also Get() method that returns `*T` alongside with const version.